torch/csrc/distributed/rpc/process_group_agent.cpp

```diff
@@ -257,6 +257,16 @@ void ProcessGroupAgent::shutdown() {
       recvWork_->abort();
     }
   }
+  // Abort any pending sends to any destination rank.
+  {
+    std::lock_guard<std::mutex> lock(pendingSendMutex_);
+    for (auto& it : currentPendingSends_) {
+      const auto& pendingSends = it.second;
+      for (auto send : pendingSends) {
+        send->abort();
```

Comment:
Do we need to check complete before abort? I assume aborting completed sends is OK?

Reply:
We should be able to `abort` completed sends, though we could save this call which requires a lock and a CV notify in gloo if we check for completed.